components/status_dropdown/status_dropdown.jsx

```diff
@@ -196,19 +196,11 @@ export default class StatusDropdown extends React.PureComponent {
         const setCustomTimedDnd = needsConfirm ? () => this.showStatusChangeConfirmation('dnd') : this.setCustomTimedDnd;
 
         const dndSubMenuItems = this.dndTimes.map((time, index) => {
-            if (index === 5) {
-                return {
-                    id: `dndTime-${time}`,
-                    direction: 'right',
-                    text: time,
-                    action: () => setCustomTimedDnd(),
-                };
-            }
             return {
-                id: `dndTime-${time}`,
+                id: `dndTime-${index}-${time.split(' ')[0]}`,
                 direction: 'right',
-                text: time,
-                action: () => setDnd(event, index),
+                text: localizeMessage('status_dropdown.dnd_sub_menu_item.time',time),
+                action: () => index === 5 ? () => setCustomTimedDnd() : () => setDnd(event, index),
```

Comment:
that way a function will be returned, that does again return a function.
just remove the `() => ` from the start:

```
action: index === 5 ? () => setCustomTimedDnd() : () => setDnd(event, index),
```